sycl/source/detail/os_util.cpp

```diff
@@ -59,6 +58,15 @@ namespace sycl {
 inline namespace _V1 {
 namespace detail {
 
+#if defined(__INTEL_PREVIEW_BREAKING_CHANGES)
+static std::string getDirName(const char *Path)
```

Comment:
BTW, why couldn't we make this function static? It is located in cpp and is not exported. why do we need __INTEL_PREVIEW_BREAKING_CHANGES here?

Reply:
`os_util.hpp`: 

```
/// Groups the OS-dependent services.
class __SYCL_EXPORT OSUtil {
```

As such, `OSUtil::getDirName` **is** exported. The preview guard is to avoid ABI breaking change and it was added before this PR (as part of my pre-C++11 ABI fixes).

Reply:
Totally, this function should be static. But, we can only do that under PREVIEW_BREAKING_CHANGES flag because that would be an ABI breaking change (OSUtils class is exported)

Reply:
> `os_util.hpp`:
> 
> ```
> /// Groups the OS-dependent services.
> class __SYCL_EXPORT OSUtil {
> ```
> 
> As such, `OSUtil::getDirName` **is** exported. The preview guard is to avoid ABI breaking change and it was added before this PR (as part of my pre-C++11 ABI fixes).

oh, I thought it is a local function. Thank you for the clarification